client/sharkey_client.go

```diff
@@ -155,12 +155,12 @@ func (c *context) enroll(hostKey string, signedCert string) {
 		return
 	}
 	defer os.Remove(tmp.Name())
-	err = os.Chmod(tmp.Name(), 0666)
+	err = os.Chmod(tmp.Name(), 0644)
 	if err != nil {
 		log.Printf("Error calling chmod on %s: %s\n", tmp.Name(), err)
 		return
 	}
-	err = ioutil.WriteFile(tmp.Name(), body, 0666)
+	err = ioutil.WriteFile(tmp.Name(), body, 0544)
```

Comment:
Don't you mean 0644 here? 0544 is executable but not writable!

Reply:
sigh, I do.

Reply:
I should add a test around this since I fat-fingered it once already.

Reply:
Yeah, a test would be great!